src/main/java/org/opentripplanner/netex/mapping/TripPatternMapper.java

```diff
@@ -138,33 +140,13 @@ TripPatternMapperResult mapTripPattern(JourneyPattern journeyPattern) {
         }
 
         List<Trip> trips = new ArrayList<>();
-        List<TripOnServiceDate> tripOnServiceDates = new ArrayList<>();
 
         for (ServiceJourney serviceJourney : serviceJourneys) {
             Trip trip = tripMapper.mapServiceJourney(serviceJourney);
 
-            if(datedServiceJourneys.containsKey(serviceJourney.getId())) {
-                for (DatedServiceJourney datedServiceJourney : datedServiceJourneys.get(
-                        serviceJourney.getId())
-                ) {
-                    var opDay = operatingDayById.lookup(
-                            datedServiceJourney.getOperatingDayRef().getRef()
-                    );
-                    if (opDay == null) {
-                        continue;
-                    }
-                    ServiceDate serviceDate =
-                            new ServiceDate(opDay.getCalendarDate().toLocalDate());
-
-                    tripOnServiceDates.add(
-                            new TripOnServiceDate(idFactory.createId(datedServiceJourney.getId()),
-                                    trip, serviceDate,
-                                    TripServiceAlterationMapper.mapAlteration(
-                                            datedServiceJourney.getServiceAlteration())
-                            )
-                    );
-                }
-            }
+            // Add the dated service journey to the model for this trip [if it exists]
+            mapDatedServiceJourney(serviceJourney, trip);
```

Comment:
I think this line should be moved down to after checking if trip is null as there are cases with our dataset where TripOnServiceDates are created with a trip that is null. Perhaps this is an indication to a problem with our NeTEx but it's something that is happening :)

Reply:
This is fixed now. Also the root cause will be fixed by https://github.com/opentripplanner/OpenTripPlanner/pull/4049